File: algorithms/Maximum Level Sum of a Binary Tree/Maximum Level Sum of a Binary Tree.cpp
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        int maxSum = INT_MIN;
        int minLevel = 1;
        
        int level = 1;
        queue<TreeNode*> qq;
        qq.push(root);
		// level order traversal
        while(!qq.empty()){
            int size = qq.size();
            int currSum  = 0;
            while(size--){
                auto node = qq.front();
                qq.pop();
                currSum += node->val;
                if(node->left)qq.push(node->left);
                if(node->right)qq.push(node->right);
            }
            if(currSum>maxSum){
                maxSum = currSum;
                minLevel = level;
            }
            level++;
        }
        
        return minLevel;
    }
};

File: algorithms/M/Maximum Level Sum of a Binary Tree/Maximum Level Sum of a Binary Tree.js
var maxLevelSum = function(root) {
    let res;
    let maxSum = -Infinity;
    let currLevel = 1;
    
    const queue = [root];
    
    while (queue.length) {
        let levelSize = queue.length;
        let levelSum = 0;
        
        for (let i = 0; i < levelSize; i++) {
            let curr = queue.shift();
            levelSum += curr.val;
            
            if (curr.left) queue.push(curr.left);
            if (curr.right) queue.push(curr.right);
        }
        
        if (maxSum < levelSum) {
            res = currLevel;
            maxSum = levelSum
        }
        
        currLevel++;
    }
    
    return res;
};
